src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -512,10 +525,12 @@ private IEnumerable<string> CollectOutputs(BuildUpToDateCheckLogger logger)
         private bool CheckOutputs(BuildUpToDateCheckLogger logger, IDictionary<string, DateTime> timestampCache)
         {
             // We assume there are fewer outputs than inputs, so perform a full scan of outputs to find the earliest
-            (DateTime? outputTime, string outputPath) = GetEarliestOutput(CollectOutputs(logger), timestampCache);
+            (DateTime? outputTime, string? outputPath) = GetEarliestOutput(CollectOutputs(logger), timestampCache);
 
             if (outputTime != null)
             {
+                Assumes.NotNull(outputPath);
```

Comment:
Would be nice if this taught the below code that `outputPath` can no longer be `null`. I don't believe there's an attribute for that either.